### PR TITLE
pre-package trackmaps with app; add timeout to trackmap downloading t…

### PR DESCRIPTION
**Pull request by brentpicasso**
Improve first time experience for users by pre-packaging track maps as defaults. 

* If no trackmaps are present, defaults are loaded. 
* Added timeout for url open to when fetching trackmap updates
* removed first time download prompt
* naming refactoring of TrackMap class UI widget
* formatting / other cleanup refactoring

addresses #902 #468

### CODE REVIEW

File: autosportlabs/racecapture/tracks/trackmanager.py [hunk 1]
```diff
@@ -309,11 +319,21 @@ def load_current_tracks_worker(self, success_cb, fail_cb, progress_cb=None):
...
             self.update_lock.release()
-        
+
+    def check_load_default_tracks(self):
+        track_file_names = os.listdir(self.tracks_user_dir)
+        if (len(track_file_names) == 0):
+            Logger.info("TrackManager: No tracks found; loading defaults")
+            try:
+                with zipfile.ZipFile(os.path.join(self.base_dir, 'defaults', 'default_tracks.zip'), 'r') as z:
+                    z.extractall(self.tracks_user_dir)
```
Comment (ryandoherty) [inline autosportlabs/racecapture/tracks/trackmanager.py]: Have we tested how long this takes on a slow tablet/phone? Just want to make sure we don't lock up things for a long time.

File: build_release_apk.sh [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 set -e
 rm -rf bin
+python -m build_tools.build_default_tracks
```
Comment (ryandoherty) [inline build_release_apk.sh]: Need any sort of checking here if this fails? Do we want to fail the build completely?

File: autosportlabs/racecapture/tracks/trackmanager.py (hunk 1 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/tracks/trackmanager.py]: On the crappy Asus reference tablet it's about a quarter second, so it should be negligible for a long time.

File: build_release_apk.sh (hunk 2 above)
Comment (brentpicasso) [inline build_release_apk.sh]: conveniently, nope.  If the python script raises an error, the bash script stops.

Comment (ryandoherty) [inline build_release_apk.sh]: :+1:

File: autosportlabs/racecapture/tracks/trackmanager.py (hunk 1 above)
Comment (ryandoherty) [inline autosportlabs/racecapture/tracks/trackmanager.py]: :+1:

File: defaults/.gitignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Ignore everything in this directory
+*
+# Except this file
+!.gitignore
```
Comment (ryandoherty) [inline defaults/.gitignore]: I'm confused why we need this gitignore. Can we use the top level one?

Comment (brentpicasso) [inline defaults/.gitignore]: it's to force the presence of the directory, and prevent anything from being checked in to this directory.

**Comment by ryandoherty** [comment]
Overall looks good and works great! A great improvement.

**Pull request merged**